unixfs/hamt/util.go

```diff
@@ -0,0 +1,61 @@
+package hamt
+
+import (
+	"math/big"
+)
+
+type hashBits struct {
```

Comment:
Would be nice to have some comments explaining what exactly the purpose of `hashBits` as a data structure is and how `.Next` works/what it returns